### PR TITLE
Rename `taurus` to `GEMD`

### PR DESCRIPTION
**Pull request by maxhutch**
This is performed in a couple steps:
 - Copy `./taurus` to `./gemd`
 - Rename `TaurusJSON` and `TaurusEncoder` to `GEMDJson` and `GEMDEncoder`
 - Split the tests for `./taurus` and `./gemd` in `.travis.yml` to avoid test class name conflicts.
 - Rename "taurus" to "GEMD" in the `README.md`
 - Add "./gemd" line to strip-hints.sh
 - Replace "taurus" with "gemd" in "docs/"
 - Add deprecation warning to "taurus/__init__.py"
 - Replace every source file (non-test) in taurus with an import from gemd
 - Remove flake8 coverage of taurus
 - Reduce testing coverage to 98% in taurus since not all modules are directly imported.
 - Bump version and edit url in setup.py

### CODE REVIEW

**Comment by maxhutch** [comment]
@kroenlein does the layout in `./gemd` look right?  I'm going to dedup all the stuff in `./taurus` yet, so don't worry about that.

**Comment by maxhutch** [comment]
> There are still several outstanding references to taurus in the docs and scripts folders.

Could you help me find them all with some inline comments?

File: README.md [hunk 1]
```diff
@@ -1,32 +1,32 @@
...
-To install `taurus`, you can simply:
+To install `gemd`, you can simply:
 ```
-$ pip install taurus-citrine
+$ pip install gemd
 ```
 
 Detailed documentation of the `GEMD` data model can be found in the [language-agnostic documentation](https://citrineinformatics.github.io/gemd-docs/).
-Documentation of this package can be found [here](https://citrineinformatics.github.io/taurus/).
+Documentation of this package can be found [here](https://citrineinformatics.github.io/gemd/).
```
Comment (kroenlein) [inline README.md]: Are we migrating the location of this repo as well, or just renaming things?

Comment (maxhutch) [reply README.md]: We will be, yes.

File: setup.py [hunk 2]
```diff
@@ -35,15 +35,15 @@ def run(self):
         develop.run(self)
 
 
-setup(name='taurus-citrine',
-      version='0.6.0',
+setup(name='gemd',
+      version='0.6.1',
```
Comment (maxhutch) [inline setup.py]: Patch version bump since this is backward compatible.

File: taurus/__init__.py [hunk 3]
```diff
@@ -1 +1,9 @@
 """Data concepts library."""
+
+import warnings
+warnings.warn("The taurus module is deprecated in favor of gemd; "
```
Comment (maxhutch) [inline taurus/__init__.py]: Here is the new deprecation warning.

**Comment by maxhutch** [comment]
After we merge this, I'll rename the repository to `CitrineInformatics/gemd`

File: gemd/demo/cake.py [hunk 4]
```diff
@@ -18,6 +18,7 @@
 from gemd.entity.util import complete_material_history, make_instance
 from gemd.entity.file_link import FileLink
 from gemd.entity.source.performed_source import PerformedSource
+from gemd.json import GEMDJson
```
Comment (sparadiso) [inline gemd/demo/cake.py]: Is this a bugfix? I was expecting a replace not add.

Comment (maxhutch) [reply gemd/demo/cake.py]: I had accidentally removed it from the copy in an earlier commit.  You can see it in the original `cake.py` [here](https://github.com/CitrineInformatics/taurus/blob/f49a2b27fe4a27f30903aa924a18b347de630858/taurus/demo/cake.py#L22)

File: taurus/util/tests/test_substitute_objects.py [hunk 5]
```diff
@@ -50,6 +50,7 @@ def func(base_ent):
     measurement = MeasurementSpec(name="name", parameters=parameter, template=meas_template)
     test_dict = {"foo": measurement}
     recursive_foreach(test_dict, func, apply_first=True)
+    print(test_dict)
```
Comment (bfolie) [inline taurus/util/tests/test_substitute_objects.py]: Should this be removed?

**Comment by bfolie** [comment]
> * Reduce testing coverage to 98% in taurus since not all modules are directly imported.

Is it possible to explicitly omit the modules that are not imported and set the coverage to 100%?

File: taurus/__init__.py (hunk 3 above)
Comment (bfolie) [reply taurus/__init__.py]: Should we specify a point at which we'll remove support for the taurus module in imports?

**Comment by maxhutch** [comment]
> > * Reduce testing coverage to 98% in taurus since not all modules are directly imported.
> 
> Is it possible to explicitly omit the modules that are not imported and set the coverage to 100%?

Its a pretty long list.  I don't think we get anything out of it, since we shouldn't be editing this code anymore anyways.

**Pull request merged**